### DISCUSSION TITLE
Recycled precipitation

### DISCUSSION THREAD

**Discussion by ViniSantos16** [WAM2layers/WAM2layers#414] [Q&A] [answered]
How should I calculate the recycled rainfall based on the model? It would be: e_track / p_track_lower + p_track_upper?

**Comment by ruudvdent** [comment, marked as answer]
More or less. In any case the brackets should be placed around the p_track variables

Note that all output is in kg m-2 s-1, so integration over time and space is essential.

So, if you do this with the output data directly the result will make no sense whatsoever

When everything is converted to kg by taking grid cell sizes into account you can make this computation

**Reply by ViniSantos16** [reply]
Thank you very much for your answer!
Do you suggest doing the calculation in another way?
For example, from the fluxes_storages data?

**Reply by ruudvdent** [reply]
no, from the output, but with further analysis steps. Not the output directly.

**Reply by XM23** [reply]
Hello everyone, I have currently used the WAM2Layers model to perform data preprocessing, forward tracking, and backward tracking. However, I have significant confusion regarding the calculation of PRR and ERR, and have encountered issues repeatedly during the calculations—with PRR even reaching 100% in some grid points.
I would like to inquire about which variables from forward and backward tracking should be selected when calculating the pixel-wise Precipitation Recycling Ratio (PRR) and Evaporation Recycling Ratio (ERR) for the study area. When computing PRR, should we use the e_track dimension/tagged_prec from the backward tracking results, the total precipitation data in e_track/preprocessed data, or the total precipitation data in tagged_prec/preprocessed data? Additionally, which dimension should be chosen for calculating the Evaporation Recycling Ratio? While reading the paper on the WAM2Layers model Version 3.0, I noticed that recycling-related functionalities will be added in the future. Are there any relevant codes available for assistance now? Thank you very much.
